pages/articles/docker.md

```diff
@@ -0,0 +1,201 @@
+
+![Using software in virtual environments, called containers](./../images/docker/dockercontainers.png)
+
+The good news is that you will not even need to install PnP.PowerShell in containers by your own: PnP team is already publishing Docker container images for each supported and nightly release, [here](https://hub.docker.com/r/m365pnp/powershell). You will however need to install docker runtime.
+
+If you use Windows, we would recommend you using Linux containers with help of WSL. Alternatively, even though it is not the most common way, you might want to run PnP.PowerShell in Windows containers. Mind you that some use cases might be limited when using Windows containers. If you use Mac OS or Linux, the easiest way is to use Linux containers.
+
+## Play with PnP.PowerShell online
+
+You can try using m365pnp/powershell Docker containers online, without installing anything on your desktop/laptop.
+
+1. Open https://labs.play-with-docker.com/
+
+2. Login
+
+3. Click `Start`
+
+4. Click `+ ADD NEW INSTANCE`
+
+5. Run in the online console:
+
+    ```
+    docker run --rm -it m365pnp/powershell
+    ```
+
+After that you can start running commands like `Connect-PnPOnline`.
+
+## Installing Docker locally
+
+\<to be verified\>
+
+- Mac OS:
+
+    1. Install
+
+    ```bash
+    brew install --cask docker
+    ```
+
+    2. Launch Docker:
+
+        - Press ⌘ + Space to bring up Spotlight Search and enter Docker to launch Docker.
+        - In the Docker needs privileged access dialog box, click OK.
+        - Enter password and click OK.
+
+- Linux:
+
+    ```bash
+    bash <(curl -s https://get.docker.com/)
+    sudo usermod -aG docker $USER
+    newgrp docker
+    ```
+
+- Linux containers on Windows (using WSL):
+
+    1. Start installing WSL:
+
+        ```powershell
+        Enable-WindowsOptionalFeature -Online -FeatureName "Microsoft-Windows-Subsystem-Linux", "VirtualMachinePlatform" -NoRestart
+        Invoke-WebRequest -Uri https://wslstorestorage.blob.core.windows.net/wslblob/wsl_update_x64.msi -OutFile wsl_update_x64.msi -UseBasicParsing
+        Start-Process msiexec.exe -Wait -ArgumentList "/I $((Get-Location).Path)\wsl_update_x64.msi /quiet"
+        ```
+
+    2. Reboot the machine before you continue
+
+    3. Finish installing WSL:
+
+        ```powershell
+        wsl --set-default-version 2
+        wsl --install --distribution Ubuntu
+        ```
+
+    4. Install docker in WSL:
+
+        ```bash
+        bash <(curl -s https://get.docker.com/)
+        sudo usermod -aG docker $USER
+        newgrp docker
+        ```
+
+- Windows:
+
+    ```powershell
+    iex ((New-Object System.Net.WebClient).DownloadString('https://chocolatey.org/install.ps1'))
+    choco install -y docker-engine
+    ```
+
+## Using PnP.PowerShell
+
+Sometimes you want to run inline PnP.PowerShell commands.
+
+- Linux/WSL/Mac OS:
+
+    \<to be verified\>
+
+    ```bash
```

Comment:
Same as above, can we please remove this section, the `to be verified` one ? If we are not sure, lets not add it and confuse users